subgraph/src/fleek-nfa.ts

```diff
@@ -257,6 +254,91 @@ export function handleMetadataUpdateWithIntValue(
   }
 }
 
+export function handleInitialized(event: InitializedEvent): void {
+    // This is the contract creation transaction.
+    log.warning('This is the contract creation transaction.', []);
+    if (event.receipt) {
+      let receipt = event.receipt as ethereum.TransactionReceipt;
+      log.warning('Contract address is: {}', [
+        receipt.contractAddress.toHexString(),
+      ]);
+ 
+      // add owner
+      let owner = new Owner(event.transaction.from);
+      owner.collection = true;
+      owner.save();
+    }
+}
+
+export function handleCollectionRoleChanged(event: CollectionRoleChangedEvent): void {
+  let toAddress = event.params.toAddress;
+  let byAddress = event.params.byAddress;
+  let role = event.params.role;
+  let status = event.params.status;
```

Comment:
```suggestion
  const { toAddress, byAddress, role, status } = event.params;
```

Reply:
Discussed on Discord with @zoruka.
Summary:

This doesn't work in the data source mapping file because of the following error:
```
ERROR TS1003: Identifier expected.

   const { tokenId, toAddress, byAddress, role, status } = event.params;
```

Seems like the compiler does not recognize the restructured attribution.

Not going to be implemented.